api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java

```diff
@@ -102,12 +102,28 @@
  * <em>Implicit</em> converters are only created when no other converter was found; therefore, they do not have a
  * priority.
  *
+ * <h3 id="empty">Empty values</h3>
+ *
+ * For all converters, the empty string {@code ""} <em>must</em> be considered an empty value. Some converters <em>may</em>
+ * consider other values to be empty as well.
+ * <p>
+ * When a conversion results in an empty value, most converters should normally return {@code null}.  All of the built-in
+ * converters <em>must</em> return {@code null} for an empty value, <em>except</em> for the built-in converters for
```

Comment:
what about the primitive types, e.g. int?

Reply:
There are no primitive type converters, only boxed type converters.  And boxed type converters must also return `null` for empty values.  If a CDI-style property has a "primitive" type, it will unbox for you, but it cannot be empty by nature (because primitive values have no empty representation).

Don't confuse conversion result with the result of the property retrieval operation.  If a conversion returns `null` and `getValue` or an equivalent (e.g. CDI injection of a non-optional type) is used, then `NoSuchElementException` would be thrown (as these operations can never yield a `null` value), because the value is empty and the type does not allow empty values.

Types such as `OptionalInt` allow an emptiable `int` value to be represented.

Reply:
I understand from the implementation point of view. I directly comment on the above text. In the converter spec, we list `int` under Built-in converters. I think people will be confused when they saw the above statement. 

> 

The following Converters are provided by MicroProfile Config by default:

    boolean and java.lang.Boolean , values for true (case insensitive) "true", "1", "YES", "Y" "ON". Any other value will be interpreted as false

    byte and java.lang.Byte

    short and java.lang.Short

    int, java.lang.Integer, and java.util.OptionalInt

    long, java.lang.Long, and java.util.OptionalLong

    float and java.lang.Float; a dot '.' is used to separate the fractional digits

    double, java.lang.Double, and java.util.OptionalDouble; a dot '.' is used to separate the fractional digits

    char and java.lang.Character

    java.lang.Class based on the result of Class.forName



Maybe in the built-in converters section, we can add the primitive convertion is done via the boxed type.